File: php/block.php
<?php

class block
{
    public $previousHash;
    public $timestamp;
    public $data;
    public $hash;
}

function addBlock($data, $blockChain) {
	$blockChain[] = new block();
	$count = count($blockChain);
	$block = $blockChain[$count - 1];
	if ($count != 1) $block->previousHash = $blockChain[$count - 2]->hash;
	$block->timestamp = time();
	$block->data = $data;
	$block->hash = md5(($count - 1) . $block->previousHash . $block->timestamp . $data);
	return $blockChain;
}

function chainVerify($blockChain) {
	$res = 0;
	for ($i = 0; $i <= count($blockChain) - 1; $i++) {
		$block = $blockChain[$i];
		$previousHash = "";
		if ($i != 0) $previousHash = $blockChain[$i - 1]->hash;
		$timestamp = $block->timestamp;
		$data = $block->data;
		if ($block->hash != md5(($i) . $previousHash . $timestamp . $data)) {
			$res = $i;
			break;
		}
	}
	return $res;
}

function chainsCompare($blockChain1, $blockChain2) {
	if (!chainVerify($blockChain1)) return 0;
	if (!chainVerify($blockChain2)) return 0;
	if (count($blockChain1) != count($blockChain2)) return 0;
	return $blockChain1[count($blockChain1) - 1]->hash == $blockChain2[count($blockChain2) - 1]->hash ? 1 : 0;
}

function printBlock($id, $blockChain) {
	$block = $blockChain[$id];
	echo "ID: $id\n";
	echo "Предыдущий хэш: $block->previousHash\n";
	echo "Дата/время: " . date('Y-m-d/H:i:s', $block->timestamp) . "\n";
	echo "Данные: $block->data\n";
	echo "Хэш: $block->hash\n";
}

function printVerify($blockChain) {
	$res = chainVerify($blockChain);
	echo $res == 0 ? "Верификация цепочки: ok\n" : "Верификация цепочки: ошибка в блоке #$res\n";
}

?>

File: python/block.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# encoding=utf8

import sys  
import time
from datetime import datetime
import hashlib

reload(sys)  
sys.setdefaultencoding('utf8')

class block:
    def __init__(self, previousHash, timestamp, data, hash):  
        self.previousHash = previousHash
        self.timestamp = timestamp
        self.data = data
        self.hash = hash

def addBlock(data, blockChain):
    count = len(blockChain)
    previousHash = "" if count == 0 else blockChain[count - 1].hash
    timestamp = int(time.time())
    hash = hashlib.md5((str(count) + previousHash + str(timestamp) + data).encode()).hexdigest()
    blockChain.append(block(previousHash, timestamp, data, hash))
    return blockChain

def chainVerify(blockChain):
    res = 0
    for i in range(0, len(blockChain)):
        block = blockChain[i]
        previousHash = ""
        if (i != 0):
            previousHash = blockChain[i - 1].hash
        timestamp = block.timestamp
        data = block.data
        if (block.hash != hashlib.md5(str(i) + previousHash + str(timestamp) + data).hexdigest()):
            res = i
            break
    return res

def chainsCompare(blockChain1, blockChain2):
    if (not chainVerify(blockChain1)):
        return 0
    if (not chainVerify(blockChain2)):
        return 0
    if (len(blockChain1) != len(blockChain2)):
        return 0
    return 1 if blockChain1[len(blockChain1) - 1].hash == blockChain2[len(blockChain2) - 1].hash else 0

def printBlock(id, blockChain):
    block = blockChain[id]
    print "ID: " + str(id)
    print "Предыдущий хэш: " + block.previousHash
    print "Дата/время: " + datetime.utcfromtimestamp(block.timestamp).strftime('%Y-%m-%d %H:%M:%S')
    print "Данные: " + block.data
    print "Хэш: " + block.hash

def printVerify(blockChain):
    res = chainVerify(blockChain)
    print "Верификация цепочки: ok" if res == 0 else "Верификация цепочки: ошибка в блоке #" + str(res)